lib/token-util.js

```diff
@@ -0,0 +1,46 @@
+var assign = require('lodash.assign')
+
+var Token // Token constructor from markdown-it
+
+var tokenUtil = module.exports = {}
+
+tokenUtil.set = function (TokenConstructor) {
+  Token = TokenConstructor
+}
+
+// set/replace or append to an attribute on a markdown-it Token
+tokenUtil.attr = function (token, attr, value, operation) {
+  var idx = token.attrIndex(attr)
+  if (idx < 0) {
+    token.attrPush([attr, value])
+  } else {
+    if (operation === 'append') {
+      token.attrs[idx][1] += ' ' + value
+    } else {
+      token.attrs[idx][1] = value
+    }
+  }
+}
```

Comment:
Once #100 is merged, this function can disappear, and our [headings plugin](https://github.com/revin/marky-markdown/blob/emoji-in-heading-anchors/lib/headings.js#L40-L43) can switch to using markdown-it 5.1's new [`attrJoin`/`attrSet` methods](https://github.com/markdown-it/markdown-it/commit/18f1531024604c090c16233ee44c572be3115ee0). @ashleygwilliams Should that happen in 7.0.1, or should I rebase this PR onto the other one so it all happens in 7.0.0?